### PR TITLE
feat: improvements with AST parsing

### LINKED ISSUE #1: Throwing Sourcemap error
I tried this plugin out today and it seems to work well, but I'm seeing the following error in terminal when it runs:
```
rendering chunks (1)...Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
rendering chunks (2)...Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
Sourcemap is likely to be incorrect: a plugin (vite-plugin-react-remove-attributes) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help
```

Did some quick Googling and found this, which may or may not be helpful? https://github.com/vitejs/vite/issues/4939

### PR DESCRIPTION
**Pull request by feslima**
@l-mbert I've changed the regex match into a node removal directly from the JS AST. 

The reason was that the AST libraries are in charge of validating the parsing (I think, but I'm not sure).

I've tested locally, and I believe this can fix #1 and #2. I haven't measured performance or anything like that, though.

Please, have a look. Any feedback is more than welcome.

Cheers.

### CODE REVIEW

**Comment by l-mbert** [comment]
For me the Example doesn't run. The Problem is that you try to parse the `index.html` File.
Could you add a check for that? Thanks!

Try changing the `vite.config.js` in the `example/` Folder to import the Plugin from `../dist` and build it before running `yarn build` in the Example Folder.

**Comment by feslima** [comment]
@l-mbert Sorry about that. I forgot the mention the `include` and `exclude` parameters, they follow [this doc](https://github.com/rollup/plugins/tree/master/packages/pluginutils#createfilter). I'm going to update the `README` file with  the pattern options to select which files are to be parsed. For the example, the `vite.config.js` file should be like this:

```javascript
import { defineConfig } from 'vite';

import VitePluginReactRemoveAttributes from '../dist/index';

export default defineConfig({
  plugins: [
    VitePluginReactRemoveAttributes({
      attributes: ['data-testid'],
      include: './src/*.(ts|tsx|js|jsx)',
    }),
  ],
});
```
If the `include` option is not defined it will consider all files to be parsed. Should I define a default value (besides updating the instructions)? If so, which one do you think would be most appropriate? Or I could make it a required parameter to be specified by the user.

**Comment by feslima** [comment]
I've made the `include` parameter required by the user when setting the plugin and updated the `README` and example `vite.config.js` files

**Comment by feslima** [comment]
Hey @l-mbert, sorry if I'm being annoying, but is this PR good enough?

**Comment by dermoumi** [comment]
I think `include` should be optional and have a default value that matches all `(c|m)?(t|j)sx?` (or `mjs|cjs|js|jsx|mts|cts|ts|tsx`)

Other than that, I think this is much safer than the current regex based removal. Hopefully this gets merged.

**Comment by feslima** [comment]
@dermoumi I'm not so familiar with the JS community, but my question is: how common is the `src` folder structure for JS projects? 

I'm not so sure matching all JS/TS files in a project would be a sensible default regardless of folder structure.

What do you think of a default like `./src/**/*.(c|m)?(j|t)sx?`

**Comment by dermoumi** [comment]
@feslima do not check for the whole path by default, only the base name. Also, probably exclude `**/node_modules/**` by default too, while at it.

`transform()` only receives files that are currently in rollup/vite pipeline; If you have a `vite.config.ts` file at the root and it's not (indirectly) imported by the entrypoint file, it will never go through the pipeline ;)

And to answer your question, it's very common to put your file in `src/` but we shouldn't count on that.

**Comment by feslima** [comment]
@dermoumi thank you for the feedback. I've set the defaults as you suggested.

I will wait a few days for @l-mbert to merge/refuse the PR. If there is no response I will create a repo/npm package because I will need this in the near future.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # vite-plugin-react-remove-attributes
 
-## !!! NOT ALL CASES TESTED YET, USE WITH CARE !!!
+## !!! NOT ALL CASES TESTED YET, USE WITH CARE
```
Comment (dermoumi) [inline README.md:3]: You didn't have to touch this, but it doesn't matter :')

**Review by dermoumi — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (feslima) [reply README.md:3]: Sorry about that, my editor config has an auto formatter for markdown files.

**Comment by l-mbert** [comment]
@feslima Sorry, was busy with moving the last Days, is merged and Published

**Pull request merged**